api/lib/src/lib.rs

```diff
@@ -382,10 +456,10 @@ pub trait BrokerApi: SignedExtrinsicApi + Sized + Send + Sync + 'static {
 			)
 		}) {
 			Ok(SwapDepositAddress {
-				address: deposit_address.to_string(),
+				address: AddressString::from_encoded_address(deposit_address),
```

Comment:
My editor is complaining about `expected &EncodedAddress, found EncodedAddress` here and below, but clippy is fine with it. Little confused about that.

Reply:
Yeah I had the same...